news/the_news/news_scrappers/news_scrappers/pipelines.py

```
@@ -0,0 +1,48 @@
+# See: http://doc.scrapy.org/en/latest/topics/item-pipeline.html
+from the_news.models import News, NewsPaper
+from datetime import datetime
+from django.db import DatabaseError
+
+
+class NewsScrappersPipeline(object):
+    news_paper = None
+
+    def open_spider(self, spider):
+        self.news_paper, created = NewsPaper.objects.update_or_create(name='The News',
+                                                                      defaults={'name': 'The News',
+                                                                                'website': 'https://www.thenews.com.pk/'})
+
+    def process_item(self, item, spider):
+        try:
+            date = datetime.strptime(item['date'].strip(), '%B %d, %Y')
+            url = item['url'].strip()
+            img_url = item['img_url'].strip()
+            abstract = item['abstract'].strip()
+            detail = item['detail']
+            title = item['title'].strip()
+
+            detailed_content = ''
+            for content in detail:
+                detailed_content += content + '\n'
+            detail = detailed_content
+        except ValueError:
+            return item
+
+        try:
+            news_item, created = News.objects.update_or_create(news_url=url,
+                                                               defaults={'title': title.encode('utf-8'),
+                                                                         'date': date,
+                                                                         'abstract': abstract.encode('utf-8'),
+                                                                         'news_url': url, 'image_url': img_url,
+                                                                         'detail': detail.encode('utf-8'),
+                                                                         'news_paper': self.news_paper})
+            news_item.save()
+        except DatabaseError:
+            pass
```

Should log a warning instead of silently passing it.